internal/command/format/diff.go

```diff
@@ -454,11 +456,22 @@ func (p *blockBodyDiffPrinter) writeNestedAttrDiff(
 	p.buf.WriteString(p.color.Color("[reset]"))
 	p.buf.WriteString(strings.Repeat(" ", nameLen-len(name)))
 
-	if old.HasMark(marks.Sensitive) || new.HasMark(marks.Sensitive) {
-		p.buf.WriteString(" = (sensitive value)")
+	// Then schema of the attribute itself can be marked sensitive, or the values assigned
+	sensitive := attrWithNestedS.Sensitive || old.HasMark(marks.Sensitive) || new.HasMark(marks.Sensitive)
+	if sensitive {
+		p.buf.WriteString(" = (sensitive")
+		if attrWithNestedS.Sensitive {
+			p.buf.WriteRune(')')
+		} else {
+			p.buf.WriteString(" value)")
+		}
 		if p.pathForcesNewResource(path) {
 			p.buf.WriteString(p.color.Color(forcesNewResourceCaption))
 		}
+
+		if new.IsNull() {
+			p.buf.WriteString(p.color.Color("[dark_gray] -> null[reset]"))
```

Comment:
If the value is sensitive, showing `null` is still disclosing information. I think `null` should be treated the same as any other value.

Reply:
I was trying to match the behavior of standard types with sensitive values, which show -> null when they are to be unset or destroyed. Is that a separate bug?

Reply:
A sensitive `null` value should still be sensitive. The output I see for basic attributes is `= (sensitive) -> (sensitive)` even is the new value is `null`. I think the only case where `null` might be currently shown is if the prior value was sensitive, but not the attribute in the schema.

The `null` `new` value in this block could itself still have a sensitive mark (which was checked in the `if` clause), or the attribute may be marked sensitive in the schema. It should only be if neither of those things are true that `null` is shown. Perhaps there were other cases which aren't checking properly, which would be a bug.